tinygrad/codegen/linearizer.py

```diff
@@ -75,7 +75,7 @@ def rename_var(v: VariableOrNum, expr: str): return v if isinstance(v, NumNode)
         (g_idx, g_valid), amt, dim = self.sts[i].expr_idxs(fake_idxs), 1, None
     else:
       g_idx, g_valid = self.sts[i].expr_idxs(fake_idxs)
-    localtype = dtypes.float32 if amt == 1 else dtypes.float.vec(amt)
+    localtype = (dtypes.float64 if isinstance(buf, (ConstBuffer, MemBuffer)) and buf.dtype == dtypes.float64 else dtypes.float32) if amt == 1 else dtypes.float.vec(amt)
```

Comment:
i didn't know where to do this check against `ConstBuffer` (needed for the `0.0`) and `MemBuffer` (needed for the `acc0`) or something else, not sure of the implications to the rest of the codebase